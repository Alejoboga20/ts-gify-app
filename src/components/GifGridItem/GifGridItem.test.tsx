import { render, screen } from '@testing-library/react';

import { GifGridItem } from '.';

const testTitle = 'test title';
const testUrl = 'https://localhost.com/testUrl.png';

describe('GifGridItem Tests', () => {
  it('should match the component snapshot', () => {
    const { container } = render(<GifGridItem title={testTitle} gifUrl={testUrl} />);
    expect(container).toMatchSnapshot();
  });

  it('show show the image with url and alt', () => {
    render(<GifGridItem title={testTitle} gifUrl={testUrl} />);
    const { src, alt } = screen.getByRole<HTMLImageElement>('img');

    expect(src).toBe(testUrl);
    expect(alt).toBe(testTitle);
  });
});
