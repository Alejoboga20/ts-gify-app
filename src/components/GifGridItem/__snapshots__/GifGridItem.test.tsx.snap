// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifGridItem Tests > should match the component snapshot 1`] = `
<div>
  <div
    class="h-full w-full rounded border border-neutral-200 p-2"
  >
    <img
      alt="test title"
      class="h-[200px] w-full"
      src="https://localhost.com/testUrl.png"
    />
  </div>
</div>
`;
